src/net_processing.cpp

```diff
@@ -1260,6 +1260,17 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             return false;
         }
 
+        if (nServices & ((1 << 7) | (1 << 5))) {
+            if (GetTime() < 1533096000) {
+                // Immediately disconnect peers that use service bits 6 or 8 until August 1st, 2018
+                // These bits have been used as a flag to indicate that a node is running incompatible
+                // consensus rules instead of changing the network magic, so we're stuck disconnecting
```

Comment:
nit: suggest change the wording of this comment to `... to indicate that a node is running incompatible consensus rules. We will disconnect based on these service bits until August 1st 2018 (when we expect the incompatible nodes to have changed their network magic).`

We're not 'stuck' doing anything. This is strictly an improvement both for nodes which have hard-forked away and those which maintain Bitcoin consensus rules.